### ISSUE TITLE
Investigate if SnowCD can be integrated with Airbyte snowflake connector

### ISSUE THREAD

**Issue by subodh1810** [airbytehq/airbyte#9933]
Network issues are tough to debug and reproduce. Specially if they happen randomly. SnowCD is a tool provided by Snowflake to diagnose and troubleshoot network connection issues. We should try to integrate it as part of Airbyte (may be as part of check_connection) and also try to run it when a snowflake connector fails.
Ref : https://docs.snowflake.com/en/user-guide/snowcd.html
This was @ChristopheDuong's idea and I think it will be super helpful

**Comment by ChristopheDuong** [comment]
Snowcd was brought up by @jrhizor!

**Comment by sherifnada** [comment]
@subodh1810 what's the relative priority on this? should this be in our current sprint? can it be handed it off to GL?

**Comment by subodh1810** [comment]
@sherifnada 

> what's the relative priority on this?

this is mainly to enable us to be better at debugging network issue. At this point we have seen a 1 instance of network issue where we are not sure whats happening. It would be nice to have when we go to GA. 

> should this be in our current sprint?

I just put it in the current sprint so that we get eyes on it. Its fine if we move to the next sprint.

> can it be handed it off to GL?

So there are two things that we are aiming for. First is integrating SnowCD in snowflake connector as part of check spec and intgerate it with normalization. Second when a snowflake connector fails either in normalization or destination, we again run SnowCD to make sure there were no network issues involved. 
I think the first part can be handled via GL to integrate it in connector and normalization but I am not sure how to trigger it when the connector fails

**Comment by ChristopheDuong** [comment]
> I think the first part can be handled via GL to integrate it in connector and normalization but I am not sure how to trigger it when the connector fails

I'm not sure if the tool would need to be run from the exact same worker pod that is encountering the network issue to be more precise or not... or if executing from a brand new pod would be enough insights.

if implemented in the `check_connection`, SnowCD could be triggered asynchronously (at a later time) by:
https://github.com/airbytehq/airbyte/issues/2591
(or synchronously in the same temporal workflow, but that would start up a new/separate worker for it anyway?)

And yes, we can delegate the snowCD implementation.

**Comment by subodh1810** [comment]
@ChristopheDuong the reason I wanted to trigger it from the same pod is cause these network failures are interim network failures and happen randomly.

**Comment by ChristopheDuong** [comment]
Yes I agree with you there @subodh1810, that's also my questioning in my comment!

it would be more helpful if it was part of the same process but that would make the implementation more specific

For example, for normalization, you could add a "if integration_type = snowflake then snowcd" in this file:
https://github.com/airbytehq/airbyte/blob/8e8f402b8a6a0cb87f4f52e906f7c5ea46eeb66e/airbyte-integrations/bases/base-normalization/entrypoint.sh#L121

but that makes it more intrusive than implementing it as a separate check_connection triggered by #2591
